application/forms/ConfigForm.php

```diff
@@ -58,7 +58,7 @@ public function onSuccess()
     {
         $sections = array();
         foreach ($this->getValues() as $sectionAndPropertyName => $value) {
-            if (empty($value)) {
+            if ($value === '' || (is_array($value) && empty($value))) {
```

Comment:
Slightly different solution: `if ($value === '' || $value === false || $value === array()) {`
`false` gets converted to `''`